sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -496,4 +520,37 @@ static long getTimeoutMillis(Duration configuredTimeout, long defaultTimeout) {
         // Return the maximum of the timeout period and the minimum allowed timeout period.
         return Math.max(configuredTimeout.toMillis(), MINIMUM_TIMEOUT);
     }
+
+    private static boolean shouldApplyProxy(SocketAddress socketAddress, Pattern nonProxyHostsPattern) {
+        if (nonProxyHostsPattern == null) {
+            return true;
+        }
+
+        if (!(socketAddress instanceof InetSocketAddress)) {
+            return true;
+        }
+
+        InetSocketAddress inetSocketAddress = (InetSocketAddress) socketAddress;
+
+        return !nonProxyHostsPattern.matcher(inetSocketAddress.getHostString()).matches();
+    }
+
+    /*
+     * Request has started, add the Azure SDK ChannelAdapter.
+     */
+    private void doOnRequest(HttpClientRequest request, Connection connection, long writeTimeout, long responseTimeout,
```

Comment:
Consider naming this `addRequestHandler` to not misinterpret this as `doOnRequest` method provided by Netty and also this describes what the method is doing.